pkgs/development/interpreters/j/default.nix

```diff
@@ -1,40 +1,42 @@
-{ lib, stdenv, fetchFromGitHub, readline, libedit, bc
+{ lib
+, stdenv
+, fetchFromGitHub
+, bc
+, libedit
+, readline
 , avxSupport ? stdenv.hostPlatform.avxSupport
 }:
 
 stdenv.mkDerivation rec {
   pname = "j";
-  version = "902";
-  jtype = "release-b";
+  version = "904-beta-c";
+
   src = fetchFromGitHub {
+    name = "${pname}-source";
     owner = "jsoftware";
     repo = "jsource";
-    rev = "j${version}-${jtype}";
-    sha256 = "0j67vgikqflwjqacsdicasvyv1k54s2c8vjgwmf0ix7l41p4xqz0";
-    name = "jsource";
+    rev = "j${version}";
+    hash = "sha256-MzEO/saHEBl1JwVlFC6P2UKm9RZnV7KVrNd9h4cPV/w=";
   };
 
-  buildInputs = [ readline libedit bc ];
-  bits = if stdenv.is64bit then "64" else "32";
-  platform =
-    if (stdenv.isAarch32 || stdenv.isAarch64) then "raspberry" else
-    if stdenv.isLinux then "linux" else
-    if stdenv.isDarwin then "darwin" else
-    "unknown";
-  variant = if stdenv.isx86_64 && avxSupport then "avx" else "";
-
-  j64x="j${bits}${variant}";
+  buildInputs = [
+    readline
+    libedit
+    bc
+  ];
 
-  doCheck = true;
-
-  # Causes build failure due to warning
-  hardeningDisable = lib.optional stdenv.cc.isClang "strictoverflow";
+  platform =
+    if (stdenv.isAarch32 || stdenv.isAarch64) then "raspberry"
+    else if stdenv.isLinux then "linux"
+    else if stdenv.isDarwin then "darwin"
```

Comment:
If this piece of code is getting moved around anyway, maybe put the Darwin condition first now? Apparently there is now some M1 support

Reply:
Okay!

Reply:
Isn't the raspberry case `isLinux`, though?

Reply:
If this is the case, then what? raspberry is irrelevant?
In any case Borg built both linux intel and arm 64 bits.

Reply:
If we have raspberry, it should be above Linux in the condition list (but maybe dropping it if aarch64 works anyway is also fine).

I would mostly keep the old logic and move Darwin up, so Darwin?ARM?Linux?

In the current expression keeping raspberry is pointless, I guess we can also try dropping it and see how it goes.

The current code just looks strange.

Reply:
Looking at the [original code](https://github.com/jsoftware/jsource/blob/4f422d11a4006c8c68db2ef815d871e88bc398e5/make2/build_all.sh#L21), it is a bit weirder. I think I will rewrite the Nix expression in order to match it.

Reply:
@7c6f434c now everything is working as expected!

Reply:
I especially like the `j64x:=j32` case (upstream).

Thanks for your work!